dali/operators/reader/loader/numpy_loader.h

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_READER_LOADER_NUMPY_LOADER_H_
+#define DALI_OPERATORS_READER_LOADER_NUMPY_LOADER_H_
+
+#include <dirent.h>
+#include <sys/stat.h>
+#include <errno.h>
+
+#include <fstream>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+#include <algorithm>
+#include <regex>
+#include <memory>
+
+#include "dali/core/common.h"
+#include "dali/pipeline/data/types.h"
+#include "dali/operators/reader/loader/file_loader.h"
+#include "dali/util/file.h"
+
+
+namespace dali {
+TypeInfo TypeFromNumpyStr(const std::string &format);
+
+class NumpyParseTarget{
+ public:
+  std::vector<int64_t> shape;
+  TypeInfo type_info;
+  bool fortran_order;
+
+  size_t size() {
```

Comment:
If you use `TensorShape<>` you can just do `volume(shape)` to get the total number of elements

Reply:
you can use `volume` on most containers.

Reply:
using volume now.